qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,521 @@
+    w = np.copy(v)
+    v[qubit_position] = 1
+    qubit_op = Pauli((v, w))
+    return SparsePauliOp(qubit_op)
+
+
+def _add_sparse_pauli(qubit_op1: SparsePauliOp, qubit_op2: SparsePauliOp) -> SparsePauliOp:
+    """
+    Return `qubit_op1` and `qubit_op2`, except when either one is `None`.
+    In the latter case, return the one that is not `None`. In other words, assume
+    `None` signifies the additive identity.
+    """
+    if qubit_op1 is None:
+        return qubit_op2
+    elif qubit_op2 is None:
+        return qubit_op1
+    else:
+        return qubit_op1 + qubit_op2
+
+
+def _to_physicist_index_order(facs: List) -> Tuple[List, int]:
+    """
+    Reorder the factors `facs` to be two raising operators followed by two lowering operators and
+    return the new factors and the phase incurred by the reordering. Note that `facs` are not in
+    chemists' order, but rather sorted by index with least index first.
+
+    Args:
+      facs: a list of factors where each element is `(i, c)` where `i` is an integer index and
+        `c` is either `-` or `+`.
+
+    Returns:
+        facs_out: A copy of the reordered factors or the input list (not a copy) if the factors are
+          already in the desired order.
+        phase: Either `1` or `-1`.
+
+    Raises:
+        ValueError: if `facs` does not represent a two-body interaction.
+    """
+    ops = [fac[1] for fac in facs]
+    if ops == ["+", "+", "-", "-"]:
+        facs_out = facs
+        phase = 1
+    elif ops == ["+", "-", "+", "-"]:
+        facs_out = [facs[0], facs[2], facs[1], facs[3]]
+        phase = -1
+    elif ops == ["+", "-", "-", "+"]:
+        facs_out = [facs[0], facs[3], facs[1], facs[2]]
+        phase = 1
+    else:
+        raise ValueError("unexpected sequence of operators", facs)
+    return facs_out, phase
+
+
+class BravyiKitaevSFMapper(FermionicMapper):
+    """The Bravyi-Kitaev super fast fermion-to-qubit mapping.
+
+    This implementation follows closely Reference [1].
+
+       References:
+           [1]: Bravyi-Kitaev Superfast simulation of electronic structure on a quantum computer
+                Kanav Setia and James D Whitfield
+                JCP Volume 148, Issue 16 - Published April 2018
+                arXiv:1712.00446
+    """
+
+    def map(self, second_q_op: FermionicOp) -> PauliSumOp:
+        if not isinstance(second_q_op, FermionicOp):
+            raise TypeError("Type ", type(second_q_op), " not supported.")
+
+        edge_list = bksf_edge_list_fermionic_op(second_q_op)
+        sparse_pauli = _convert_operators(second_q_op, edge_list)
+
+        ## Simplify and sort the result
+        sparse_pauli = sparse_pauli.simplify()
+        indices = sparse_pauli.table.argsort()
+        table = sparse_pauli.table[indices]
+        coeffs = sparse_pauli.coeffs[indices]
+        sorted_sparse_pauli = SparsePauliOp(table, coeffs)
+
+        return PauliSumOp(sorted_sparse_pauli)
+
+
+def _analyze_term(term_str: str) -> Tuple[str, List]:
```

Comment:
Can we either put the main class of this file all the way at the front or end? Right now, it seems to be arbitrarily sandwiched between utility methods...

Reply:
done